### ISSUE TITLE
eval_gegenbauer inconsistently supporting limit evaluation

### ISSUE THREAD

**Issue by jacksonvanover** [scipy/scipy#11713]
Overall, I'm observing non-uniform support for the evaluation of the end behavior of these functions.

- limits approaching positive and negative infinity for alpha=0, n>1 return NaNs when they should return 0
- for degree n>1, limits approaching positive infinity are correctly evaluated whereas those approaching negative infinity consistently return NaN. Theoretically, evaluation of these limits should be easy to implement (with the caveat that I haven't done any poking around myself): those of even-degree should be symmetrical about the y-axis and those of odd-degree should be symmetrical about the origin.
#### Reproducing code example:

```
>>> for n in range(0,5):
...     for alpha in range(0,5):
...             print(f"n={n}; alpha={alpha}: {(special.eval_gegenbauer(n, alpha, np.NINF), special.eval_gegenbauer(n, alpha, np.inf))}")
... 
n=0; alpha=0: (1.0, 1.0)
n=0; alpha=1: (1.0, 1.0)
n=0; alpha=2: (1.0, 1.0)
n=0; alpha=3: (1.0, 1.0)
n=0; alpha=4: (1.0, 1.0)
n=1; alpha=0: (nan, nan)
n=1; alpha=1: (-inf, inf)
n=1; alpha=2: (-inf, inf)
n=1; alpha=3: (-inf, inf)
n=1; alpha=4: (-inf, inf)
n=2; alpha=0: (nan, nan)
n=2; alpha=1: (nan, inf)
n=2; alpha=2: (nan, inf)
n=2; alpha=3: (nan, inf)
n=2; alpha=4: (nan, inf)
n=3; alpha=0: (nan, nan)
n=3; alpha=1: (nan, inf)
n=3; alpha=2: (nan, inf)
n=3; alpha=3: (nan, inf)
n=3; alpha=4: (nan, inf)
n=4; alpha=0: (nan, nan)
n=4; alpha=1: (nan, inf)
n=4; alpha=2: (nan, inf)
n=4; alpha=3: (nan, inf)
n=4; alpha=4: (nan, inf)
```

#### Scipy/Numpy/Python version information:
```
1.4.1 1.17.4 sys.version_info(major=3, minor=6, micro=8, releaselevel='final', serial=0)
```

**Comment by mdhaber** [comment]
> Theoretically, evaluation of these limits should be easy to implement (with the caveat that I haven't done any poking around myself): those of even-degree should be symmetrical about the y-axis and those of odd-degree should be symmetrical about the origin.

Yes, I think so, if this means that those of even degree are even functions and those of odd degree are odd functions.

The problem exists for large finite values, too, though, so I think rather than special-casing the evaluation of infinite `x`, it can be fixed for all `x` by:
- adding a special case for `alpha = 0` to always return `0` (except possibly for `n = 0`)
- using the even/odd property of the function regardless of `x`, e.g.

```python3
    sign = -1 if ((x < 0) and (alpha % 2 == 1)) else 1
    x = abs(x)
    # multiply the final result by `sign`
```

**Comment by jorenham** [comment]
> * adding a special case for `alpha = 0` to always return `0` (except possibly for `n = 0`)

$C_0^{(0)}(x) = 0$ [according to wolfram alpha](https://www.wolframalpha.com/input?i=GegenbauerC%5B0%2C+0%2C+x%5D), but `eval_gegenbauer` currently returns `1.0` or `nan`, depending on whether `n` is int or float:

```pycon
>>> from scipy.special import eval_gegenbauer
>>> eval_gegenbauer(0, 0, [-1, 0, 1])
array([1., 1., 1.])
>>> eval_gegenbauer(0.0, 0, [-1, 0, 1])
array([nan, nan, nan])
```
